File: flow/script/types/video.ts
/**
 * Domain types for video/subtitle processing.
 */

export interface VideoMeta {
  id: string;
  title: string;
  url: string;
  index: number;
  duration?: number;
}

export interface SubtitleEntry {
  index: number;
  start: string;
  end: string;
  text: string;
}

export interface CourseConfig {
  name: string;
  playlistUrl: string;
  source: "youtube" | "bilibili";
  lang: string;
}

File: flow/script_py/models/video.py
"""
Domain types for video/subtitle processing.
"""

from dataclasses import dataclass
from typing import Optional


@dataclass
class VideoMeta:
    id: str
    title: str
    url: str
    index: int
    duration: Optional[float] = None


@dataclass
class SubtitleEntry:
    index: int
    start: str
    end: str
    text: str


@dataclass
class CourseConfig:
    name: str
    playlist_url: str
    source: str  # "youtube" | "bilibili"
    lang: str
